lib/openstudio-standards/btap/carbon/btap_carbon.rb

```diff
@@ -92,6 +92,23 @@ def audit_embodied_carbon
 
             # Note that the carbon tallying method must be called per-surface
             # to account for the perimeter values for each surface.
+
+            # rd2: Reiterating a note in envelope_costing.rb:
+            #
+            #      For NECB 2017, 2020, etc., a surface-specific construction's RSi
+            #      is the end result of TBD 'derating' calculations based on linear
+            #      thermal bridging - done strictly for OpenStudio modelling
+            #      purposes. These end-of-the-line, surface-specific 'derated'
+            #      construction RSi values are unsuitable for BTAP's costing or
+            #      embodied carbon calculations, as they do not reflect the initial
+            #      'uprated' (real-world) clear-field constructions. Better to rely
+            #      on TBD-reported 'uprated' Uo-factors.
+            #
+            #      In btap/bridging.rb, I added (to each 'derated' surface in the
+            #      OSM) an AdditionalProperty ("uprated_Uo"). This provides
```

Comment:
If costing and embodied carbon calculations are to be based strictly on _post-simulation_ OSM file content (not _live_ BTAP class attributes), then maintaining surface-specific "uprated_Uo" _AdditionalProperty_ instances seems like a straightforward solution.